### ISSUE TITLE
Question about Skforecast classes: ForecasterDirectMultiVariate vs ForecasterRecursiveMultiSeries

### ISSUE THREAD

**Issue by valdilemke** [skforecast/skforecast#977]
Hello,

First of all, I would like to thank you for the excellent work in developing the Skforecast library. It has been extremely helpful in my experiments with time series.

I am working with a dataset from a poultry farm, which contains the following features:

- Age (in days)
- Maximum Temperature
- Minimum Temperature
- Feed Consumption
- Water Consumption
- Production Percentage (% Production)

The dataset includes records from 3 production cycles, with 2 cycles used for training and 1 for testing. The goal is to forecast the next day’s % Production (step = 1), using the other attributes as predictors.

I have tested different window sizes (lags), ranging from 1 to 14, and obtained good results using the ForecasterDirectMultiVariate class. However, I have a question about the forecasting logic used by this approach.

My questions:

About the step (horizon) parameter:
I am using step=1 to forecast one point at a time (the next day), but I perform forecasts for around 400 days ahead (in the test set). In this case, when predicting the second point in the test set, is the data from the first test point already used as input? Or does the class exclusively use training data for all predictions? I’m confused because the documentation shows a figure indicating that ForecasterDirectMultiVariate always uses only the last points from the training set. Should I train a model with step=400 instead?

About the ForecasterAutoregRecursiveMultiSeries class:
Is this class intended only for independent time series? I tested it with my dataset, but the results were quite distorted.

I would appreciate guidance on which approach would be more suitable for my context. I’m exploring different forecasting strategies with sliding windows as part of the experiments for my dissertation.

Thank you in advance for your attention and support, and I apologize for the lengthy message.

**Comment by valdilemke** [comment]
Good morning, Javier!
Thank you again for your reply.

I would like to clarify a doubt regarding the behavior of the forecast in my use case.

My dataset consists of time-ordered items indexed from 1 to 1385. I am using the first 922 items for training. My goal is to generate forecasts from item 923 up to item 1385.

Here is a visual representation of my setup:

![Image](https://github.com/user-attachments/assets/2e374d8b-1596-4091-8428-3c35d317c476)

See below how my code is currently:

```
modelos = {
    'Ridge': Ridge(random_state=123),
    'RandomForest': RandomForestRegressor(random_state=42),
    'MLP': MLPRegressor(random_state=42),
    'XGB': XGBRegressor(random_state=123)
}
tamanhos_janela = [1, 2, 3, 4, 5, 7, 14]

for janela in tamanhos_janela:
        lags = [janela]
        steps = 1

        cv = TimeSeriesFold(
            steps=steps,
            initial_train_size=len(df_train),
            refit=False,
            allow_incomplete_fold=True
        )

        for modelo_nome, modelo in modelos.items():
            print(f'Base: {base_nome} | Janela: {janela} | Modelo: {modelo_nome}')


            forecaster = ForecasterDirectMultiVariate(
                regressor=modelo,
                level=target,
                lags=lags,
                steps=steps,
                transformer_series=StandardScaler(),
                transformer_exog=None,
                weight_func=None,
                window_features=None,
                n_jobs='auto'
            )

            forecaster.fit(series=df_train.drop(columns=['Lote']).reset_index(drop=True))

            metrics_modelo, predictions = backtesting_forecaster_multiseries(
                forecaster=forecaster,
                series=df.drop(columns=['Lote']).reset_index(drop=True),
                metric=metrics,
                n_jobs='auto',
                verbose=False,
                show_progress=False,
                cv=cv
            )
```

Would the behavior, in this case, be as simulated in the image below?

![Image](https://github.com/user-attachments/assets/a0d588df-49e4-4513-91d6-027ef03a4251)

Or would it be this one?

![Image](https://github.com/user-attachments/assets/7dc98119-adf2-433c-882d-3a592ebdba66)

**Comment by JavierEscobarOrtiz** [comment]
Hello @valdilemke,

Since `refit` is set to `False` in the `TimeSeriesFold`, you are performing the second one, "backtesting no refit". For the first option, you need to use `refit=True` and `fixed_train_size=True`.

One recommendation, you don't need to fit your forecaster before the backtesting, it is done inside the function.

Javier
